core/src/main/java/org/mule/runtime/core/internal/exception/GlobalErrorHandler.java

```diff
@@ -41,56 +34,69 @@ public Publisher<CoreEvent> apply(Exception exception) {
 
   @Override
   public void initialise() throws InitialisationException {
-    setFromGlobalErrorHandler();
-    super.initialise();
+    if (!reuseGlobalErrorHandler()) {
+      super.initialise();
+      return;
+    }
+
+    if (!initialised.getAndSet(true)) {
+      setFromGlobalErrorHandler();
+      super.initialise();
+    }
```

Comment:
is this change relative to the original code or the one with the partial change already done?